File: UnitTest.cpp
#include <iostream>
#include "Solution.hpp"

class UnitTest {
private:
    Solution sol;
    vector<pair<vector<int>, bool>> testcases;
public:
    UnitTest() {
        testcases = {{{3,4,5,1,2}, true}, {{2,1,3,4}, false}, {{1,2,3}, true}};
    }

    void test() {
        for(int i = 0; i < testcases.size(); ++i) {
            vector<int>& nums = testcases[i].first;
            bool output = testcases[i].second;
            bool result = sol.check(nums);
            cout << "TestCase " << i+1 << ": " << ((result == output) ? "passed":"result") << endl;
        }
    }
};

int main() {
    UnitTest test;
    test.test();
}

File: UnitTest.py
from Solution import Solution
from timeout_decorator import timeout
import unittest

class UnitTest(unittest.TestCase):
    def setUp(self):
        self.__testcases = {1: ([3,4,5,1,2], True), 2: ([2,1,3,4], False), 3: ([1,2,3], True)}
        self.__sol = Solution()
        return super().setUp()
    
    @timeout(0.5)
    def test_case_1(self):
        nums, output = self.__testcases[1]
        result = self.__sol.check(nums = nums)
        self.assertIsInstance(result, bool)
        self.assertEqual(result, output)

    @timeout(0.5)
    def test_case_2(self):
        nums, output = self.__testcases[2]
        result = self.__sol.check(nums = nums)
        self.assertIsInstance(result, bool)
        self.assertEqual(result, output)

    @timeout(0.5)
    def test_case_3(self):
        nums, output = self.__testcases[3]
        result = self.__sol.check(nums = nums)
        self.assertIsInstance(result, bool)
        self.assertEqual(result, output)

if __name__ == '__main__': unittest.main()